tools/contrib/fsurdat_modifier.py

```diff
@@ -0,0 +1,356 @@
+                    action = "store", 
+                    dest = "fsurdat_in",
+                    type = str,
+                    default = "/glade/p/cesmdata/cseg/inputdata/lnd/clm2/surfdata_map/release-clm5.0.18/surfdata_0.9x1.25_hist_78pfts_CMIP6_simyr2000_c190214.nc")
+        parser.add_argument('--fsurdat_out', 
+                    help = 'Output surface dataset. [default: %(default)s]', 
+                    action = "store", 
+                    dest = "fsurdat_out",
+                    type = str,
+                    default = "/glade/scratch/" + myname + "/surfdata_0.9x1.25_hist_78pfts_CMIP6_simyr2000_c190214_modified.nc")
+
+        return parser
+
+
+def get_git_sha():
+    """
+    Returns Git short SHA for the currect directory.
+    """
+    return subprocess.check_output(['git', 'rev-parse', '--short', 'HEAD']).strip().decode() 
+
+
+class ModifyFsurdat:
+    """
+    A case to encapsulate function that modifies fsurdat file.
+
+    ...
+
+    Attributes
+    ----------
+
+    Methods
+    -------
+    add_tag_to_filename(filename, tag)
+       add a tag and timetag to a filename ending with 
+       [._]cYYMMDD.nc or [._]YYMMDD.nc 
+
+    update_metadata(nc, filename)
+
+    modify:
+        Modify input surface dataset
+    """
+
+    def __init__(self, overwrite_single_pft, dominant_pft,
+                 zero_nonveg_landunits, uniform_snowpack, no_saturation_excess):
+        super().__init__()
+        self.overwrite_single_pft = overwrite_single_pft
+        self.dominant_pft = dominant_pft
+        self.zero_nonveg_landunits = zero_nonveg_landunits
+        self.uniform_snowpack = uniform_snowpack
+        self.no_saturation_excess = no_saturation_excess
+
+
+    @staticmethod
+    def add_tag_to_filename(filename, tag):
+        """
+        Add a tag and replace timetag of a filename
+        # Expects file to end with [._]cYYMMDD.nc or [._]YYMMDD.nc
+        # Add the tag to just before that ending part
+        # and change the ending part to the current time tag
+        """
+
+        basename = os.path.basename(filename)
+        cend = -10
+
+        if ( basename[cend] == "c" ):
+           cend = cend - 1
+        if ( (basename[cend] != ".") and (basename[cend] != "_") ):
+           print ( "Trouble figuring out where to add tag to filename:" + filename )
+           os.abort()
+
+        today = date.today()
+        today_string = today.strftime("%y%m%d")
+
+        return(basename[:cend] + "_" + tag + "_c" + today_string + '.nc')
+
+
+    @staticmethod
+    def update_metadata(nc, filename):
+
+        #update attributes
+        today = date.today()
+        today_string = today.strftime("%Y-%m-%d")
+
+        #get git hash
+        sha = get_git_sha()
+
+        nc.attrs['Created_on'] = today_string
+        nc.attrs['Created_by'] = myname
+        nc.attrs['Created_with'] = os.path.abspath(__file__) + " -- " + sha
+        nc.attrs['Created_from'] = filename
```

Comment:
- [ ] To be announced soon: The new required metadata for inputdata files will be the following; "N/A" can be put in place for anything not applicable (e.g., contact in this case?):
  - title
  - summary (Plain English descriptive short one sentence, maybe two)
  - creator (Name only, no email)
  - contact (e.g., CAM Subboard at https://bb.cgd.ucar.edu)
  - creation_date (e.g., 2019-11-21)
  - data_script (script or instructions used to generate the dataset)
  - description (anything else relevant)
    - Maybe capturing the command-line here would be good (sys.argv)